src/apprt/gtk-ng/class/dialog.zig

```diff
@@ -0,0 +1,69 @@
+const std = @import("std");
+const adw = @import("adw");
+const gobject = @import("gobject");
+const gtk = @import("gtk");
+
+const gresource = @import("../build/gresource.zig");
+const adw_version = @import("../adw_version.zig");
+const Common = @import("../class.zig").Common;
+const Config = @import("config.zig").Config;
+
+const log = std.log.scoped(.gtk_ghostty_dialog);
+
+/// Dialog is a simple abstraction over the `adw.AlertDialog` and
+/// `adw.MessageDialog` widgets, chosen at comptime based on the linked
+/// Adwaita version.
+///
+/// Once we drop support for Adwaita <= 1.2, this can be fully removed
+/// and we can use `adw.AlertDialog` directly.
+pub const Dialog = extern struct {
+    const Self = @This();
+    parent_instance: Parent,
+
+    pub const Parent = if (adw_version.supportsDialogs())
+        adw.AlertDialog
+    else
+        adw.MessageDialog;
+
+    pub const getGObjectType = gobject.ext.defineClass(Self, .{
+        .name = "GhosttyDialog",
+        .classInit = &Class.init,
+        .parent_class = &Class.parent,
+    });
+
+    pub fn present(self: *Self, parent: ?*gtk.Widget) void {
+        switch (Parent) {
+            adw.AlertDialog => self.as(adw.Dialog).present(parent),
+            adw.MessageDialog => self.as(gtk.Window).present(),
```

Comment:
Note that this currently doesn't that as expected on Adw 1.2. How parenting works on 1.2 is that since all MessageDialogs are also windows, their parents must also be windows and be set via `gtk.Window.setTransientFor` (normally `adw.MessageDialog.new` already does this for you).

I think the easiest thing to do here is to change the parent type to `?*gtk.Window` and then setting the transient parent window immediately before presenting. It's a bit ugly but it should work

Reply:
In the MessageDialog's constructor the `transient-for` property is explicitly set (in Zig code the equivalent would be `gobject.ext.newInstance(MessageDilog, .{ .@"transient-for" = parent })`), so when you subclass it that property won't be set. GTK is one of those strange systems where the `new` method and the `init` method aren't the same thing, so subclasses don't recursively call the constructors of parent classes when properties are involved like that.

Reply:
https://github.com/ghostty-org/ghostty/blob/main/src/apprt/gtk/CloseDialog.zig#L38

The old apprt has an extra parameter passed to `adw.MessageDialog.new`

Also yeah it's totally okay to not set it or set it to null, I'm just saying that in case it does makes sense to set a parent (e.g. command palette) then we do need to specify it ourselves